### PR TITLE
Salvors

### PR DESCRIPTION
**Pull request by SamBrownMT**

### CODE REVIEW

File: backoffice/src/Data/DroitsContext.cs [hunk 1]
```diff
@@ -106,6 +107,21 @@ protected override void OnModelCreating(ModelBuilder modelBuilder)
             entity.Property(e => e.LastModified);
             entity.Property(e => e.Type);
         });
+        
+        modelBuilder.Entity<Salvor>(entity =>
+        {
+            entity.ToTable("salvors");
+
+            entity.Property(e => e.Id);
```
Comment (slkendell-mt) [inline backoffice/src/Data/DroitsContext.cs:115]: Very small, but can you change the var to be s => s pls

Comment (slkendell-mt) [reply backoffice/src/Data/DroitsContext.cs:115]: I've done it for the other entities on my current branch

File: backoffice/src/Models/Domain/AddressDetails.cs [hunk 2]
```diff
@@ -0,0 +1,24 @@
+using System.ComponentModel.DataAnnotations.Schema;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+
+namespace Droits.Models;
```
Comment (slkendell-mt) [inline backoffice/src/Models/Domain/AddressDetails.cs:5]: can you move this in folder & namespace to Droits.Entities.Models

File: backoffice/src/Models/FormModels/SalvorForm.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+
+namespace Droits.Models.FormModels;
+
+public class SalvorForm
+{
+    public SalvorForm()
+    {
+    }
+
+    public SalvorForm(Salvor salvor)
```
Comment (slkendell-mt) [inline backoffice/src/Models/FormModels/SalvorForm.cs:13]: Okay a bit of feedback on this. 

We should have an AddressForm that is re-usable, then set with Address= new AddressForm(salvor.Address)

The Address fields are then encapsulated in the AddressForm and allows us to re-use it elsewhere

Comment (slkendell-mt) [reply backoffice/src/Models/FormModels/SalvorForm.cs:13]: See DroitForm & inner WreckForm

File: backoffice/src/Models/ViewModels/SalvorView.cs [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+namespace Droits.Models.ViewModels;
+
+public class SalvorView
+{
+    public SalvorView()
+        {
+        }
+    
+        public SalvorView(Salvor salvor)
```
Comment (slkendell-mt) [inline backoffice/src/Models/ViewModels/SalvorView.cs:13]: Likewise for the View, SalvorView should have an AddressView for re-use

File: backoffice/src/Views/Salvor/Edit.cshtml [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+                </div>
+                
+                <div class="form-group">
+                    @Html.LabelFor(m => m.DateOfBirth, new { @class = "col-form-label" })
+                    @Html.EditorFor(m => m.DateOfBirth, new { htmlAttributes = new { @class = "form-control"} })
+                    @Html.ValidationMessageFor(m => m.DateOfBirth,"", new { @class = "text-danger" })
+                </div>
+                
+                @{
+                    await Html.RenderPartialAsync("_SalvorAddressFormFields",Model);
```
Comment (slkendell-mt) [inline backoffice/src/Views/Salvor/Edit.cshtml:43]: Great use of the partial here! 

We should have this as just _AddressFormFields, Model.Address when we have the AddressForm Address on the SalvorForm

Actually it's slightly different, see DroitForm

It'll be like 

<partial name="_AddressFormFields" for="Address"/>

This basically pre-pends the inner object names onto the inputs

File: backoffice/src/Models/FormModels/SalvorForm.cs (hunk 3 above)
Comment (SamBrownMT) [reply backoffice/src/Models/FormModels/SalvorForm.cs:13]: Haven't looked at the other forms yet, but my IDE wasn't happy with Address being the property name as it shares it with a class so have set it to _Address but this looks ugly, thoughts?

File: backoffice/src/Models/FormModels/AddressForm.cs [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+        AddressTown = address.Town;
+        AddressCounty = address.County;
+        AddressPostcode = address.Postcode;
+    }
+    public string AddressLine1 { get; set; } = string.Empty;
+    public string AddressLine2 { get; set; } = string.Empty;
+    public string AddressTown { get; set; } = string.Empty;
+    public string AddressCounty { get; set; } = string.Empty;
+    [Required(ErrorMessage = "Postcode is required")]
+    [RegularExpression("([Gg][Ii][Rr] 0[Aa]{2})|((([A-Za-z][0-9]{1,2})" +
```
Comment (slkendell-mt) [inline backoffice/src/Models/FormModels/AddressForm.cs:25]: This Regex looks a bit OTT. 

^(?i)([A-Z]{1,2}[0-9]{1,2}[A-Z]?)\s?([0-9][A-Z]{2})$


Should work

File: backoffice/src/Models/FormModels/SalvorForm.cs (hunk 3 above)
Comment (slkendell-mt) [reply backoffice/src/Models/FormModels/SalvorForm.cs:13]: AddressForm?

File: backoffice/src/Views/Salvor/Index.cshtml [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+        </thead>
+        <tbody>
+        @foreach (var salvor in Model)
+        {
+            <tr>
+                <td class="p-3 text-uppercase">
+                    <a asp-action="View" asp-route-id="@salvor.Id">@salvor.Name</a>
+                </td>
+                <td class="p-3">@salvor.DateOfBirth.ToString("dd/MM/yyyy")</td>
+                <td class="p-3">@salvor.AddressPostcode</td>
```
[the remark this answers is not in the document]
Comment (SamBrownMT) [reply backoffice/src/Views/Salvor/Index.cshtml]: Added postcode there to make sure fields where saving and showing properly

File: backoffice/src/Models/FormModels/AddressForm.cs (hunk 6 above)
Comment (SamBrownMT) [reply backoffice/src/Models/FormModels/AddressForm.cs:25]: Regex taken from https://github.com/stemount/gov-uk-official-postcode-regex-helper
which says it is 'Official'

**Review by slkendell-mt — APPROVED** [review]

**Pull request merged**